Terminal.Gui/ConsoleDrivers/V2/MainLoop.cs

```diff
@@ -142,6 +144,21 @@ internal void IterationImpl ()
         Logging.IterationInvokesAndTimeouts.Record (swCallbacks.Elapsed.Milliseconds);
     }
 
+    private View? _lastTop;
+    private void HackLayoutDrawIfTopChanged ()
+    {
+        // TODO: This fixes closing a modal not making its host (below) refresh
+        //       until you click.  This should not be the job of the main loop!
```

Comment:
You're right. That must be an `Application` job.

Reply:
Somehow, it seems to work with the original drivers but not the v2 ones.

V2 drivers only listen to these properties (needs layout/draw).  There are no manual or triggered draws, so possibly that is why this hack is needed for now.

Reply:
I understand, but I think that if you do the same thing in the `Application.LayoutAndDraw` method you'll will obtain the same result, but I could be wrong.

Reply:
In my code, layout and draw is only called if a subview of Top wants redrawn or layed out... I will look into it some more later